docs/sections/papers/deita.md

```diff
@@ -0,0 +1,530 @@
+```
+
+Again the authors provided several prompts you could use to make your
+response evolve according to some guidelines. For example, this one was
+used to enrich the answer:
+
+```
+PROMPT = """
+    I want you to act as a Response Rewriter
+    Your goal is to enhance the quality of the response given by an AI assistant
+    to the #Given Prompt# through rewriting.
+    But the rewritten response must be reasonable and must be understood by humans.
+    Your rewriting cannot omit the non-text parts such as the table and code in
+    #Given Prompt# and #Given Response#. Also, please do not omit the input
+    in #Given Prompt#.
+    You Should enhance the quality of the response using the following method:
+    Please make the Response more in-depth
+    You should try your best not to make the #Rewritten Response# become verbose,
+    #Rewritten Response# can only add 10 to 20 words into #Given Response#.
+    ‘#Given Response#’, ‘#Rewritten Response#’, ‘given response’ and ‘rewritten response’
+    are not allowed to appear in #Rewritten Response#
+    #Given Prompt#:
+    <instruction_0>
+    #Given Response#:
+    <response_0>
+    #Rewritten Response#:
+    """
+```
+
+Prompting this to an LLM, you will automatically get a more enriched
+response, called response\_1, from an initial response response\_0 and
+initial instruction instruction\_0:
+
+```python
+evol_response_quality = EvolQuality(
+    name="evol_response_quality",
+    llm=OpenAILLM(model="gpt-3.5-turbo"),
+    num_evolutions=5,
+    store_evolutions=True,
+    include_original_response=True,
+    input_mappings={
+        "instruction": "evolved_instruction",
+        "response": "answer",
+    },
+    pipeline=pipeline,
+)
+
+evol_response_quality.load()
+
+_evolved_responses = next(evol_response_quality.process([{"instruction": PROMPT + instruction_0, "response": reponse_0}]))
+
+print("Original Response:")
+print(reponse_0)
+print("\nEvolved Response:")
+print(*_evolved_responses[0]['evolved_responses'], sep="\n")
+```
+
+And now, as in EVOL COMPLEXITY you iterate through this path and use
+different prompts to make your responses more relevant, helpful or
+creative. In the paper, they make 4 more iterations to get 5 evolved
+responses (R0,R1, R2, R3, R4 ) which makes 5 different responses for one
+initial instruction at the end of this step.
+
+```python
+response_quality_scorer = QualityScorer(
+    name="response_quality_scorer",
+    llm=OpenAILLM(model="gpt-3.5-turbo"),
+    input_mappings={
+        "instruction": "evolved_instruction",
+        "responses": "evolved_responses",
+    },
+    pipeline=pipeline,
+)
+
+expand_evolved_responses = ExpandColumns(
+    name="expand_evolved_responses",
+    columns=["evolved_responses", "scores"],
+    output_mappings={
+        "evolved_responses": "evolved_response",
+        "scores": "evol_response_score",
+    },
+    pipeline=pipeline,
+)
+
```

Comment:
This is not used here right? It's used in another subsequent step I believe

Reply:
Yes. I have included the data wrangling steps in line because it seemed more logical that adding them all at the end. What do you think?